internal/storage/blob/cloner_test.go

```diff
@@ -34,8 +35,7 @@ func TestCloneResult(t *testing.T) {
 
 		bucketDir := filepath.Join(dir, "bucket")
 		require.NoError(t, os.MkdirAll(bucketDir, perm775))
-
-		cacheDir := filepath.Join(dir, dotcache)
+		cacheDir := filepath.Join(dir, dotcache, base64.URLEncoding.EncodeToString([]byte(bucketDir)))
```

Comment:
It's worth creating a function to generate the cache directory name given the work directory and the bucket URL. We are currently repeating the logic in multiple places and they can slowly drift apart with time.